src/core/tables.js

```diff
@@ -0,0 +1,163 @@
+// @ts-check
+// Module core/tables
+// Handles tables in the document.
+// Generates a Table of Tables wherever there is a #tot element.
+
+import {
+  addId,
+  getIntlData,
+  renameElement,
+  showWarning,
+  wrapInner,
+} from "./utils.js";
+import { html } from "./import-maps.js";
+
+export const name = "core/tables";
+
+const localizationStrings = {
+  en: {
+    list_of_tables: "List of Tables",
+    table: "Table ",
+  },
+  ja: {
+    table: "図 ",
+    list_of_tables: "図のリスト",
+  },
+  ko: {
+    table: "그림 ",
+    list_of_tables: "그림 목록",
+  },
+  nl: {
+    table: "Figuur ",
+    list_of_tables: "Lijst met figuren",
+  },
+  es: {
+    table: "Figura ",
+    list_of_tables: "Lista de Figuras",
+  },
+  zh: {
+    table: "图 ",
+    list_of_tables: "规范中包含的图",
+  },
+  de: {
+    table: "Abbildung",
+    list_of_tables: "Abbildungsverzeichnis",
+  },
+};
+
+const l10n = getIntlData(localizationStrings);
+
+export function run() {
+  const tot = collectTables();
+
+  // Create a Table of Tables if a section with id 'tot' exists.
+  const totElement = document.getElementById("tot");
+  if (tot.length && totElement) {
+    decorateTableOfTables(totElement);
+    totElement.append(
+      html`<h1>${l10n.list_of_tables}</h1>`,
+      html`<ul class="tot">
+        ${tot}
+      </ul>`
+    );
+  }
+}
+
+/**
+ * process all tables
+ */
+function collectTables() {
+  /** @type {HTMLElement[]} */
+  const tot = [];
+  document.querySelectorAll("table").forEach((table, i) => {
```

Comment:
Let's run over `"table.numbered"` only and avoid internal checks later on.

Reply:
Ah, this PR also allows referencing unnumbered tables. Never mind above check then.

Reply:
I misunderstood why we would want tables that aren't numbered.

Tabular data (like what will show up in the list of tables) will all be numbered.
Only tables used for non-tabular data are unnumbered. But they shouldn't be linkable or included in the list of tables.
So I agree that enumerating "table.numbered" is the better option.